### PR TITLE
Add cargo-fuzz targets

### PR DESCRIPTION
**Pull request by cgm616**
This pull request add three fuzzing targets: one on `pest/meta` and one each on the grammars in `pest/grammars`. See issue #296.

### CODE REVIEW

**Comment by CAD97** [comment]
🎉

Is it possible you could add a quick "how-to" for fuzzing to README or elsewhere? (Personally I'm on Windows so I'm pretty out of luck, but it's definitely useful to have a quick "here's how to". Also it serves as a subtle "hey we fuzz the library".)

**Comment by cgm616** [comment]
For sure! I'll update the readme with how to get started as well as some strange bugs with the fuzzing interface.

**Comment by dragostis** [comment]
Does it make sense to have this information in `CONTRIBUTING.md`?

File: grammars/fuzz/Cargo.toml [hunk 1]
```diff
@@ -0,0 +1,26 @@
+
+[package]
+name = "pest_grammars-fuzz"
+version = "0.0.1"
```
[the remark this answers is not in the document]
Comment (cgm616) [reply grammars/fuzz/Cargo.toml]: This is autogenerated by the cargo-fuzz tool; I can certainly change it, but I think it makes sense to stay with what the tool generated.

File: meta/fuzz/Cargo.toml [hunk 2]
```diff
@@ -0,0 +1,22 @@
+
+[package]
+name = "pest_meta-fuzz"
+version = "0.0.1"
```
[the remark this answers is not in the document]
Comment (cgm616) [reply meta/fuzz/Cargo.toml]: Same as up above; this is auto-generated.

File: grammars/src/lib.rs [hunk 3]
```diff
@@ -17,6 +17,8 @@ extern crate pest;
 #[macro_use]
 extern crate pest_derive;
 
+pub use pest::Parser;
```
[the remark this answers is not in the document]
Comment (cgm616) [reply grammars/src/lib.rs]: The pub export of pest::Parser in grammars/src/lib.rs is to allow it to
be used by the fuzzing code, which was having trouble using the same version
of pest and reconciling the traits.

File: grammars/fuzz/Cargo.toml (hunk 1 above)
[the remark this answers is not in the document]
Comment (dragostis) [reply grammars/fuzz/Cargo.toml]: Sure. I'm fine with that. As I said, I think it only makes sense in cases where the auto-generated files are not touched anymore, but I don't have any strong opinion on it.

File: grammars/src/lib.rs (hunk 3 above)
[the remark this answers is not in the document]
Comment (dragostis) [reply grammars/src/lib.rs]: I understand why. I'm just wondering whether this is a good approach. Maybe `pest_grammars` should completely hide the `parse` function behind its own abstraction.

File: grammars/fuzz/Cargo.toml (hunk 1 above)
[the remark this answers is not in the document]
Comment (cgm616) [reply grammars/fuzz/Cargo.toml]: Let's go for it, and change it. While I'm at it, I'll go make a pull request on cargo-fuzz to change their default.

File: grammars/src/lib.rs (hunk 3 above)
[the remark this answers is not in the document]
Comment (cgm616) [reply grammars/src/lib.rs]: I think either way you get roughly the same result: a parsing library that parses json and toml without requiring a direct dependency on `pest`. If you think not leaking `Parser` is cleaner, then that can be changed.

**Comment by cgm616** [comment]
@dragostis What would the intent be if it was included in `CONTRIBUTING.md`?

**Comment by dragostis** [comment]
I guess I was thinking about having a paragraph on how fuzzing works for people looking to get involved with the project, but this is not necessarily the best fit for this info.

**Comment by cgm616** [comment]
It seems that `README.md` is mostly information for consumers of the crate, which also doesn't seem like a very good fit.

**Comment by dragostis** [comment]
Exactly. The fuzzing crates could have this information. Also, one of the could be a symlink to the other so that we don't have to maintain two versions.

**Comment by cgm616** [comment]
`README`s in the fuzzing crates describing their use is a good idea. I'm confused about the symlink suggestion; could you elaborate?

**Comment by dragostis** [comment]
For example, `pest_meta`'s `README.md` file could simply be a symlink to the one in `pest_grammars`. That way it would show up on GitHub in both directories.

**Comment by CAD97** [comment]
I think the best idea might be a `FUZZING.md` in the root directory, then symlink that as `README.md` in the fuzz target directories.

**Comment by cgm616** [comment]
I agree @CAD97, that seems to cover both bases.

**Pull request merged**